src/simsopt/_core/optimizable.py

```diff
@@ -1030,8 +1030,11 @@ def local_x(self, x: RealArray) -> None:
         if self.local_dof_size != len(x):
             raise ValueError
         self._dofs.free_x = x
-        if self.local_dof_setter is not None:
-            self.local_dof_setter(self, list(self.local_full_x))
+        
```

Comment:
@mbkumar we forgot to update the dofs on the cpp side, so here's an attempted fix